lib/functions.sh

```diff
@@ -43,12 +43,19 @@ read_stdin_command_and_verify_signature() {
 
     # this will read from standard input of the service, the data should be
     # considered untrusted
-    tr -d '\r' | awk -b \
-            -v in_command=0 \
+    LC_ALL=C head -c 65535 |
+    LC_ALL=C tr '\r\0' '\001\001' |
+    LC_ALL=C awk -b \
+            -v in_command=1 \
```

Comment:
Care to explain why? Start of the command should be considered inside the signed block, not at the beginning of the data. Anything before or after signed block should not be considered at all.

Reply:
Whoops!  As an aside, the command itself needs some more validation.  Is an awk regex a good approach?